node/src/database/db_migrations.rs

```diff
@@ -227,23 +253,17 @@ impl DbMigratorReal {
         Ok(())
     }
 
-    fn is_target_version_reached(current_state: &usize, target_version: &usize) -> bool {
-        match current_state.cmp(&target_version){
-            Less => false,
-            Equal => true,
-            Greater => panic!("Nonsensical: the version specified as a target version is lower than the mismatched one.")
-        }
-    }
-
     fn select_updates_to_process<'a>(
         mismatched_schema: usize,
         list_of_updates: &'a [&'a (dyn DatabaseMigration + 'a)],
+        target_version: usize,
         mig_utils: &dyn DBMigrationUtilities,
     ) -> Vec<&'a (dyn DatabaseMigration + 'a)> {
-        mig_utils.panic_if_the_version_found_is_too_high(mismatched_schema);
+        mig_utils.too_high_found_schema_will_panic(mismatched_schema);
         list_of_updates
             .iter()
             .skip_while(|entry| entry.old_version() != mismatched_schema)
+            .filter(|entry| entry.old_version() < target_version)
```

Comment:
Maybe take_while...but it would be silly to withhold approval because of this...